### ISSUE TITLE
VSCode formatter

### ISSUE THREAD

**Issue by pokey** [tweag/topiary#655]
Cool project! It would be great to be able to use topiary formatters for autoformatting in VSCode. See [this guide](https://code.visualstudio.com/api/language-extensions/programmatic-language-features#format-source-code-in-an-editor) for more, in case you haven't seen that one yet 😊

**Comment by Xophmeister** [comment]
Thanks for your suggestion :slightly_smiling_face: 

From this guide, it looks like native VSCode integration would be through an LSP. Topiary doesn't currently provide any LSP interface, nor is it on the radar. Can integration be done without resorting to an LSP? (@torhovland IIRC, you mentioned something about this many months ago; perhaps in the context of a VSCode extension.)

**Comment by torhovland** [comment]
I haven't looked into it in detail, but in any case there needs to be an extension involved.

**Comment by pokey** [comment]
Yes it's possible to do it without resorting to an LSP; see the subsection called **Direct Implementation** underneath the section I sent you

You might find it easiest to use a wasm rather than native code. We originally tried to build a native extension for Cursorless VSCode, and it was a nightmare to deal with the OS / arch matrix within electron context so switched to wasm and haven't looked back
